cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -238,6 +238,16 @@ func (p *PnpmLockfile) ResolvePackage(workspacePath turbopath.AnchoredUnixPath,
 		return Package{Key: key, Version: version, Found: true}, nil
 	}
 
+	if entry, ok := p.Packages[resolvedVersion]; ok {
+		var version string
+		if entry.Version != "" {
```

Comment:
Again, just to clarify, when would we have a version in the entry? If it's a git dependency would there be a version at all?

Reply:
The test data added to this PR came from pnpm which does include a version in a git dependency entry. In the case of the git dependency we rely on the key changing as the commit hash is part of the key. This will catch a dependency change since we use both the key and version when determining if a package changed.